compiler/semexprs.nim

```diff
@@ -1613,10 +1613,17 @@ proc semDefined(c: PContext, n: PNode, onlyCurrentScope: bool): PNode =
   # we replace this node by a 'true' or 'false' node:
   result = newIntNode(nkIntLit, 0)
   if not onlyCurrentScope and considerQuotedIdent(c.config, n[0], n).s == "defined":
-    if n.sons[1].kind != nkIdent:
-      localError(c.config, n.info, "obsolete usage of 'defined', use 'declared' instead")
-    elif isDefined(c.config, n.sons[1].ident.s):
-      result.intVal = 1
+    case n.sons[1].kind
+    of nkIdent:
+      if isDefined(c.config, n.sons[1].ident.s):
+        result.intVal = 1
+    of nkSym:
+      # the argument to `defined` may be of kind nkSym, if there exists e.g.
+      # a proc with the same name
+      if isDefined(c.config, n.sons[1].sym.name.s):
+        result.intVal = 1
+    else:
+      discard
```

Comment:
what if someone type something like `defined(123)`, wouldn't it better to tell the user that it is not allowed rather than let it go silently?

Reply:
With the change @Araq proposed, that is taken care of, if I'm not mistaken. :)